### PR TITLE
fix(relay): Do not expose breakdowns config to external Relays

### PR DESCRIPTION
**Pull request by jan-auer**
While Breakdowns are being developed and only processing Relays can handle breakdowns, we should not expose the configuration to external Relays. This patch moves breakdown config into the "full" configuration section, which is only emitted for trusted processing Relays (the last layer).

Alongside this change, I took the liberty to restructure tests slightly. There is no need for a test matrix on `test_get_project_config` since there's only a single case where breakdowns should be emitted at the moment.

### CODE REVIEW

**Review by untitaker — APPROVED** [review]

File: src/sentry/relay/config.py [hunk 1]
```diff
@@ -162,15 +162,12 @@ def get_project_config(project, full_config=True, project_keys=None):
...
-        breakdowns_config = project.get_option("sentry:breakdowns")
-        if breakdowns_config is not None:
-            cfg["config"]["breakdowns"] = breakdowns_config
-
     if not full_config:
         # This is all we need for external Relay processors
         return ProjectConfig(project, **cfg)
 
+    if features.has("organizations:performance-ops-breakdown", project.organization):
+        cfg["config"]["breakdowns"] = project.get_option("sentry:breakdowns")
```
Comment (dashed) [inline src/sentry/relay/config.py:170]: @jan-auer @untitaker Just for my understanding, what is considered trusted Relays?

Comment (dashed) [reply src/sentry/relay/config.py:170]: Does this new placement ensure this breakdown config will only go out to SaaS Relays?

**Review by dashed — APPROVED** [review]

File: src/sentry/relay/config.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/relay/config.py:170]: yes exactly and that's also definition of trusted relays

idk where pop relays stand, but it shouldn't matter for this PR

Comment (jan-auer) [reply src/sentry/relay/config.py:170]: Trusted Relays are meant to handle traffic across all organizations within our infrastructure, which comprises PoPs and processing Relays.

**Comment by jan-auer** [comment]
@dashed was the breakdowns feature enabled in onpremise?

**Comment by dashed** [comment]
@jan-auer It's not enabled for onpremise as it's still an early adopters feature.

**Pull request merged**